nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -753,14 +755,61 @@ export async function findUsableCommitmentsMutex(
   );
 }
 
+/**
+ *
+ * @function saveCommitments save a list of commitments in the database
+ * @param {[]} listOfCommitments a list of commitments to be saved in the database
+ * @returns if all the commitments in the list already exists in the database
+ * throw an error, else return a success message.
+ */
+export async function saveCommitments(listOfCommitments) {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+
+  /**
+   * 1. listOfCommitments => get only the ids
+   */
+  const commitmentsIds = listOfCommitments.map(commitment => commitment._id);
+  /**
+   * 2. Find commitments that already exists in DB
+   */
+  const commitmentsFound = await db
+    .collection(COMMITMENTS_COLLECTION)
+    .find({ _id: { $in: commitmentsIds } })
+    .toArray();
+
+  /**
+   * 3. remove the commitments found in the database from the list
+   */
+  const onlyNewCommitments = listOfCommitments.filter(
+    commitments =>
+      commitmentsFound.find(commitmentFound => commitmentFound.id === commitments.id) === undefined,
+  );
```

Comment:
Singular!
Very nice chain of array methods :blush: :clap: 

```suggestion
  const onlyNewCommitments = listOfCommitments.filter(
    commitment =>
      commitmentsFound.find(commitmentFound => commitmentFound.id === commitment.id) === undefined,
  );
```

Reply:
Changed!!